File: pe-core/src/main/java/org/saar/core/common/particles/ParticlesMeshWriter.java
package org.saar.core.common.particles;

import org.saar.core.mesh.writers.InstancedArraysMeshWriter;

public class ParticlesMeshWriter implements InstancedArraysMeshWriter<ParticlesVertex, ParticlesInstance> {

    private final ParticlesMeshPrototype prototype;

    public ParticlesMeshWriter(ParticlesMeshPrototype prototype) {
        this.prototype = prototype;
    }

    @Override
    public void writeVertex(ParticlesVertex vertex) {
    }

    @Override
    public void writeInstance(ParticlesInstance instance) {
        this.prototype.getPositionBuffer().getWriter().write3f(instance.getPosition3f());
        this.prototype.getBirthBuffer().getWriter().writeInt(instance.getBirth());
    }

}

File: pe-core/src/main/kotlin/org/saar/core/common/particles/ParticlesMeshWriter.kt
package org.saar.core.common.particles

import org.saar.core.mesh.writer.InstancedMeshWriter
import org.saar.lwjgl.util.DataWriter

class ParticlesMeshWriter(
    private val positionWriter: DataWriter,
    private val birthWriter: DataWriter,
) : InstancedMeshWriter<ParticlesInstance> {

    override fun writeInstance(instance: ParticlesInstance) {
        this.positionWriter.write3f(instance.position3f)
        this.birthWriter.writeInt(instance.birth)
    }
}